model/wfmodel/StateObject.php

```diff
@@ -71,14 +68,25 @@ public function getAccessRights(User $user)
         }
         return $rights;
     }
-    
-    public function __toPhpCode() {
-        return 'new '.get_class($this).'('.
-            \common_Utils::toPHPVariableString($this->id).','.
-            \common_Utils::toPHPVariableString($this->label).','.
-            \common_Utils::toPHPVariableString($this->readRoles).','.
-            \common_Utils::toPHPVariableString($this->writeRoles).','.
-            \common_Utils::toHumanReadablePhpString($this->transitions).
+
+    public function getReadRoles(): array
+    {
+        return $this->readRoles;
+    }
+
+    public function getWriteRoles(): array
+    {
+        return $this->writeRoles;
+    }
```

Comment:
I want to mention, that we in constructor we don't have types for these variables, so we cannot be sure, that they will return an array.

```php
class A
{
	private $a;

	public function __construct($a = [])
	{
		$this->a = $a;
	}

	public function getA(): array
	{
		return $this->a;
	}
}

$instance = new A(null);
$instance->getA(); // <-------- will fail
```

I think, that for now nobody is storing other types inside these properties, but for safety we can add some check like (or something like this):
```suggestion
    public function getReadRoles(): array
    {
        return is_array($this->readRoles) ? $this->readRoles : [];
    }

    public function getWriteRoles(): array
    {
        return is_array($this->writeRoles) ? $this->writeRoles : [];
    }
```

But it's not mandatory

Reply:
Agreed on this and updated :). i would like to remark that if we go PHPSTANish it would be nice to include some PHPSTAN checks, as i have seems quite a significant amount of this on the code. Maybe in the same way we deal with PHPCS? (only files affected by commit?)